x/profile/internal/types/msgs.go

```diff
@@ -0,0 +1,218 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// ----------------------
+// --- MsgCreateProfile
+// ----------------------
+
+// MsgCreateProfile defines a CreateProfile message
+type MsgCreateProfile struct {
+	Moniker  string         `json:"moniker"`
+	Name     string         `json:"name,omitempty"`
+	Surname  string         `json:"surname,omitempty"`
+	Bio      string         `json:"bio,omitempty"`
+	Pictures *Pictures      `json:"pictures,omitempty"`
+	Creator  sdk.AccAddress `json:"creator"`
+}
+
+// NewMsgCreateProfile is a constructor function for MsgCreateProfile
+func NewMsgCreateProfile(name string, surname string, moniker string, bio string, pictures *Pictures,
+	creator sdk.AccAddress) MsgCreateProfile {
+	return MsgCreateProfile{
+		Moniker:  moniker,
+		Name:     name,
+		Surname:  surname,
+		Bio:      bio,
+		Pictures: pictures,
+		Creator:  creator,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgCreateProfile) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgCreateProfile) Type() string { return ActionCreateProfile }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgCreateProfile) ValidateBasic() error {
+	if msg.Creator.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", msg.Creator))
+	}
+
+	if len(msg.Name) != 0 && len(msg.Name) < MinNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot be less than %d characters", MinNameSurnameLength))
+	}
+
+	if len(msg.Name) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Surname) != 0 && len(msg.Surname) < MinNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot be less than %d characters", MinNameSurnameLength))
+	}
+
+	if len(msg.Surname) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile surname cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(strings.TrimSpace(msg.Moniker)) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "Profile moniker cannot be blank or empty")
+	}
+
+	if len(msg.Moniker) > MaxMonikerLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile moniker cannot exceed %d characters", MaxMonikerLength))
+	}
+
+	if len(msg.Bio) > MaxBioLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile biography cannot exceed %d characters", MaxBioLength))
+	}
+
```

Comment:
Probably we should also add a check to both profile and cover pictures making sure that, if they exist, they are valid URIs.

Reply:
Yeah, I left them without any checks cause we didn't discuss about them before.
Could be possible for someone to add a picture in form of `hex(byte(pic.jpg))`?

Reply:
@bragaz Absolutely not :joy: It would make the profiles HUUUGE. They should do the same as the multimedia posts: 

1. Upload the images to IPFS
2. Link the images here